src/cpp/server/server_context.cc

```diff
@@ -173,6 +173,14 @@ void ServerContext::AddTrailingMetadata(const grpc::string& key,
   trailing_metadata_.insert(std::make_pair(key, value));
 }
 
+void ServerContext::TryCancel() const {
+  grpc_call_error err = grpc_call_cancel_with_status(
+      call_, GRPC_STATUS_CANCELLED, "Cancelled on the server side", NULL);
+  if (err != GRPC_CALL_OK) {
```

Comment:
Yes, it shouldn't. Actually, this is the reason why I had a log message there (would be interested in knowing if it returned any other status). I will change it to GPR_ERROR.